design/variadic_config.md

```diff
@@ -0,0 +1,520 @@
+Using named keys rather that YAML arrays is intentional so that when an element is removed from an array, we can disambiguate whether an item was removed from all of the subsequent elements of the array being modified.
+
+Usage examples provided in the Detailed Design section.
+
+### `reapeatable` Config Items
+
+The purpose of adding a `repeatable` attribute to Config Items is to add the capability *EXTEND* resources.
+
+The existing Config Item concept will be augmented with a new property `repeatable` to indicated the value will be an array of values rather than a scalar. The value types will still inherit from the `type` field.
+
+Config Items will also now include a `template` property to allow specifying the YAML document or sub-document to copy for this array of values.  
+
+### `reapeatable` Config Groups
+
+The purpose of adding a `repeatable` attribute to Config Groups is to add the ability to *COPY* collections of resources/config.
+
+The existing Config Group concept will be augmented with a new property `repeatable` to indicated the values in each Config Item will be an array. 
+
+Config Groups will also now include a `template` property to allow specifying the YAML document or sub-document to clone for each of these groups.  
+
+## Detailed Design
+
+While the design considered here is presented in an interleaved fashion, this proposal suggests that work be broken up in the following tasking:
+1. Repeatable Config Items
+1. Repeatable Config Groups
+
+The first consideration is how the revised API will look to Vendors using these features in their application.
+
+### Example Revised Kotskind Resources 
+
+Vendors will use the revised Config Spec to define repeatable Config Groups and Items. 
+Values are inserted by the Kots and returned as part of the API for creating the ConfigValues spec.
+Below is a representative resource that will be used for several examples.
+
+#### Config
+```yaml
+apiVersion: kots.io/v1beta1 
+kind: Config 
+metadata: 
+  creationTimestamp: null 
+  name: config-sample 
+spec: 
+  # UNCHANGED
+  groups: 
+  - name: nginx_settings 
+    title: Nginx Configs 
+    description: Config to serve as an example for creating your own
+    items: 
+    - name: "nginx_port"
+      type: "text"
+      title: "Nginx port"
+      default: "80"
+      value: ""
+
+    # NEW!
+    # This is a repeatable Config Item
+    - name: "static_files"
+      type: file
+      title: "Static Assets"
+      repeatable: true      # NEW! Tells the UI/Kots to expect an array
+      minimumCount: 3       # NEW! Validation 
+      templates:            # NEW! This desclares how the array of values will be used
+        - apiVersion: v1    # By targeting a YAML path, we can clone this node for each element of the item.
+          kind: Deployment
+          name: my-deploy
+          namespace: myApp
+          yamlPath: spec.template.spec.volumes[0].projected.sources
```

Comment:
Generated files will not have a static name to target here.  Additional options could be provided as a way to target a file.
`namePrefix: ` could be used to target generated files starting with the prefix string.  This would be used instead of name, and validation should ensure only one field is provided.

Additionally, for manifests that do not have a name (IE jobs), this template could use Selectors to find the config.  This would be a larger lift and should be in a separate feature PR.